recipes/csvmonkey/all/conanfile.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+from conans.errors import ConanInvalidConfiguration
+from conans import ConanFile, CMake, tools
+
+required_conan_version = ">=1.36.0"
+
+class CSVMONEKYConan(ConanFile):
+    name = "csvmonkey"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Header-only vectorized, lazy-decoding, zero-copy CSV file parser "
+    topics = ("csv parser", "header only", "vectorized")
+    homepage = "https://github.com/dw/csvmonkey/"
+    settings = "arch", "compiler"
+    no_copy_source = True
+    options = {"with_spirit": [True, False]}
+    default_options = {"with_spirit": False}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.arch not in ("x86", "x86_64",):
+            raise ConanInvalidConfiguration("{} requires x86 architecture.".format(self.name))
+
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("{} doesn't support Visual Studio C++.".format(self.name))
+
+    def requirements(self):
+        if self.options.with_spirit:
+            self.requires("boost/1.77.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def package(self):
+        self.copy("LICENSE*", "licenses", self._source_subfolder)
+        self.copy("*.hpp", dst="include", src=os.path.join(self._source_subfolder, "include"))
+
+    def package_id(self):
+        self.info.settings.clear()
```

Comment:
You can do `header_only` (and clear `self.info.options` as well).
This is because the resulting package does not depend on settings or options:
- The `source` method simply downloads,
- `build` does not exist
- `package` simply downloads.

If you would have been conditionally copying files depending on the value of some option, then you shouldn't remove it.
 

```suggestion
        self.info.header_only()
```